File: day01/R/Solution2.R
nums <- scan("input.txt", sep="\n")

findSolution <- function(list) {
  for (i in list) {
    for (j in list) {
      for (k in list) {
        if (i + j + k == 2020) {
          return(i * j * k)
        }
      }
    }
  }
}

print(findSolution(nums))

File: day01/python/Solution2.py
#more optimized than part 1 solution

with open('input.txt', 'r') as data:
  lines = data.read().splitlines()
  lines = [ int(x) for x in lines ]


for x in range(len(lines)):
  for y in range(x, len(lines)):
    for z in range(y, len(lines)):
      if lines[x] + lines[y] + lines[z] == 2020:
        print(lines[x] * lines[y] * lines[z])
